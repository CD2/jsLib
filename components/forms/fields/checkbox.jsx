import React from 'react'
import PropTypes from 'prop-types'

export default class Checkbox extends React.Component {

  static propTypes = {
    onChange: PropTypes.func,
    onRawChange: PropTypes.func,
  }

  handleChange = (e) => {
    const { onRawChange, onChange } = this.props
    if (onRawChange) onRawChange(e)
    if (onChange) onChange({name: e.target.name, value: e.target.checked})
  }

  render() {
    return (
      <input {...this.props} onChange={this.handleChange}/>
    )
  }
}
