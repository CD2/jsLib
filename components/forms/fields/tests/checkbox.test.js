/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { checkbox } from '../checkbox'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const wrapper = shallow(<checkbox />)
it(`snapshot`, ()=>expect(toJson(wrapper)).toMatchSnapshot())
