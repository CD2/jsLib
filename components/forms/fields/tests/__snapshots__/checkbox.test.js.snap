// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<checkbox /> snapshot 1`] = `<xheckbox />`;
